_docs/rfc/imports.md

```diff
@@ -0,0 +1,963 @@
+blocks and attributes (e.g `inputs`) is not inherited to the child config. This was something that was not possible to
+do with `include`.
+
+#### Why can't we use the merge function?
+
+In the above example, we have to explicitly declare the block. A natural extension of this pattern is to use the `merge`
+function directly on the `remote_state`:
+
+```hcl
+remote_state = merge(
+  import.root.remote_state,
+  {
+    config = {
+      key = relpath(import.root.terragrunt_dir, get_terragrunt_dir())
+    }
+  }
+)
+```
+
+Unfortunately, this is a syntax error because `remote_state` is a block, and not an attribute. You can not directly set
+blocks in this way.
+
+
+
+
+## Alternatives
+
+### In consideration
+
+#### Single terragrunt.hcl file per environment
+
+This implementation introduces a new block `module` that replaces `dependency`, `terraform`, and `inputs`. This approach
+is documented in [#759](https://github.com/gruntwork-io/terragrunt/issues/759).
+
+In addition to the general analysis of that proposal, here are the list of Pros and Cons related to the problem of
+sharing config:
+
+Pros:
+
+- Circumvent the reference problem by having all the references in a single shared scope.
+
+Cons:
+
+- If you need to share configurations across environments, you would still need one of the alternative approaches to
+  import the other config. This, however, is likely to be a rare occurrence.
+- This is a drastic change to Terragrunt, completely changing the way it works.
+
+
+Let's walk through how each of the import use cases look like with this implementation:
+
+**Hierarchical variables**
+
+In this approach, a hierarchy of variables is unnecessary because all the blocks are defined in a single scope. However,
+depending on the scope of `locals`, certain things like reusing a repetitive variable becomes more challenging.
+
+To understand this, consider a use case where you are operating in two regions, `us-east-1` and `eu-west-1`. To simplify
+this example, consider a limited application deployment where you have two modules: `vpc` and `app`.
+
+If we assume that `locals` are scoped within the file that they are defined, then you can implement this by separating
+the environment `terragrunt.hcl` into two files, `us_east_1.hcl` and `eu_west_1.hcl`:
+
+us_east_1.hcl
+
+```hcl
+locals {
+  region = "us-east-1"
+}
+
+module "vpc" {
+  # additional args omitted for brevity
+  aws_region = local.region
+}
+
+module "app" {
+  # additional args omitted for brevity
+  aws_region = local.region
+}
+```
+
+eu_west_1.hcl
+
+```hcl
+locals {
+  region = "eu-west-1"
+}
+
+module "vpc" {
+  # additional args omitted for brevity
+  aws_region = local.region
+}
+
+module "app" {
+  # additional args omitted for brevity
+  aws_region = local.region
+}
+```
+
+However, if the namespace of `locals` was shared across the environment, then the above approach would not work and each
+region file will need to define a different name for the region variable.
```

Comment:
Yea, if you're deploying the same things across regions, you'd use a different local for each region.